### PR TITLE
Fixed: The application crashes when attempting to open saved notes in the absence of the ZIM file.

### LINKED ISSUE #3682: Application crashes when opening saved notes if the ZIM file does not exist.
**Describe the bug**
The application crashes when opening saved notes if the ZIM file does not exist. 

**Expected behavior**
Instead of crashing it should not open the note and show a toast message that the zim file is not found like it is showing when we are trying to open the article.

**Steps to reproduce the behavior:**
1. Save note.
2. Delete the zim file.
3. Open the Notes screen.
4. Click on the saved note and then click `Open Note` you will see the crash.

**Environment**
- Version of Kiwix Android: 3.9.1
- Device: Any device
- OS version: Any OS

**Logs**

```kotlin
java.lang.IllegalStateException: Fragment AddNoteDialog{e13576b} (1d1493c1-8edd-4faa-9c39-30468b1a7f4f tag=AddNoteDialog) did not return a View from onCreateView() or this was called before onCreateView().                                                                                             
```

```kotlin
FATAL EXCEPTION: main
Process: org.kiwix.kiwixmobile, PID: 3807
kotlin.UninitializedPropertyAccessException: lateinit property articleNoteFileName has not been initialized
at org.kiwix.kiwixmobile.core.main.AddNoteDialog.displayNote(AddNoteDialog.kt:414)
at org.kiwix.kiwixmobile.core.main.AddNoteDialog.onViewCreated(AddNoteDialog.kt:285)
at androidx.fragment.app.Fragment.performViewCreated(Fragment.java:3128)
at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:552)
at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:261)
at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1899)
at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1817)
at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1760)
at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:547)
at android.os.Handler.handleCallback(Handler.java:958)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loopOnce(Looper.java:205)
at android.os.Looper.loop(Looper.java:294)
at android.app.ActivityThread.main(ActivityThread.java:8194)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
```

### PR DESCRIPTION
**Pull request by MohitMaliFtechiz**
Fixes #3682 

* Now, the user can view the notes even if the ZIM file does not exist. However, they cannot edit the note since the corresponding ZIM file is not present. Nevertheless, the user can still share and delete that note.
* Removed the `closeKeyboard()` method from `onFailureToCreateAddNoteDialog()` because we are calling this in our `onCreate` method and the dialog is not visible right now and this method requires the view, so it is throwing an error `this was called before onCreateView()`.

https://github.com/kiwix/kiwix-android/assets/34593983/903fe633-9932-4120-a7eb-d0b314f6e331

### CODE REVIEW

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.kt [hunk 1]
```diff
@@ -134,12 +134,15 @@ class AddNoteDialog : DialogFragment() {
       // Corresponds to "ZimFileName" of "{External Storage}/Kiwix/Notes/ZimFileName/ArticleUrl.txt"
       articleNoteFileName = getArticleNoteFileName()
       zimNotesDirectory = "$NOTES_DIRECTORY$zimNoteDirectoryName/"
+    } else {
+      // Hide the dialog if ZIM file is not found
+      // see https://github.com/kiwix/kiwix-android/issues/3682
```
Comment (gouri-panda) [inline core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.kt:139]: This line is unnecessary. We can search through the git history.

**Review by gouri-panda — COMMENTED** [review]
>Now, if the ZIM file is not found, we hide the AddNoteDialog since the selected ZIM file is unavailable, making it unnecessary to edit the note.

@MohitMaliDeveloper I'm in the favor of showing the Notes whether the corespnding zim files are deleted or not. I think users want to see the save notes too.

cc @kelson42

**Comment by kelson42** [comment]
Agree with Gouri and not really happy that the fix does not match the issue. Like written in the issue, tje note should be visible, but an error should be displayed when article tries to be open.

**Comment by MohitMaliFtechiz** [comment]
> Agree with Gouri and not really happy that the fix does not match the issue. Like written in the issue, the note should be visible, but an error should be displayed when article tries to be open.

@kelson42 I think here is some confusion, In the ticket I had purposed to hide the note and show the error toast. Because there is already a code written to hide the note if the ZIM file is not found so I was following the same approach. However, I agree with you @gouri-panda I am making the changes.

https://github.com/kiwix/kiwix-android/blob/3b4f58b229d2b12d27fb16babe9070542185b9de/core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.kt#L168-L172

https://github.com/kiwix/kiwix-android/blob/3b4f58b229d2b12d27fb16babe9070542185b9de/core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.kt#L140

**Comment by kelson42** [comment]
What is important is that the not stays visible if we display all notes.

**Comment by MohitMaliFtechiz** [comment]
@gouri-panda I have made the changes according to your review and updated the PR description according to this change.

**Comment by kelson42** [comment]
@MohitMaliDeveloper It's always appreciated if you upload a screencast in PR description when you change the UI... at least by me.

**Comment by MohitMaliFtechiz** [comment]
@kelson42 I have uploaded the screencast in the PR description.

**Review by gouri-panda — APPROVED** [review]
@MohitMaliDeveloper Thanks 🎉 . Tests would be better ;)

**Pull request merged**